base_serializer.go

```diff
@@ -83,12 +83,30 @@ func (bs *baseSerializer) sum(b []byte) []byte {
 
 func (bs *baseSerializer) handleFieldName(ionValue hashValue) error {
 	if bs.depth > 0 && ionValue.IsInStruct() {
-		fieldName := ionValue.getFieldName()
+		// TODO: Remove type cast once FieldNameSymbol is available for ion.Writer/hashWriter
```

Comment:
I think this is the same dependency as the other comment in the code. Can we open an issue if this is still pending (and not part of another PR) 

Further down in this PR I see `FieldNameSymbol` implemented so does this section of the PR need an update?

Reply:
The TODOs here are handled now.